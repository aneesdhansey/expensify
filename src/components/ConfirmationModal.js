import React from 'react'
import Modal from 'react-modal'

const ConfirmationModal = ({ isOpen, modalTitle, modalBody, handleYes, handleNo }) => (
    <Modal
        isOpen={isOpen}
        ariaHideApp={false}
        contentLabel={modalTitle}
        closeTimeoutMS={200}
        className="modal">
            <h3 className="modal__title">{modalTitle}</h3>
            <p className="modal__body">{modalBody}</p>
            <div className="modal__actions">
                <button className="button button-yes" onClick={handleYes}>Yes</button>
                <button className="button button-no button--secondary" onClick={handleNo}>No</button>
            </div>
    </Modal>
)

export default ConfirmationModal
