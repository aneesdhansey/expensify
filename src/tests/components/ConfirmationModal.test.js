import React from 'react'
import { shallow } from 'enzyme'

import ConfirmationModal from '../../components/ConfirmationModal'

let handleYes, handleNo, wrapper;

beforeEach(() => {
    handleYes = jest.fn();
    handleNo = jest.fn();
    wrapper = shallow(<ConfirmationModal 
                                isOpen={true}
                                modalTitle="Testing Modal"
                                modalBody="This is a test"
                                handleYes={handleYes}
                                handleNo={handleNo} />);
});

test('it should display ConfirmationModal correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('it should call handleYes on clicking "Yes" button', () => {

    wrapper.find('button.button-yes').simulate('click');
    expect(handleYes).toHaveBeenCalled();
});

test('it should call handleNo on clicking "No" button', () => {

    wrapper.find('button.button-no').simulate('click');
    expect(handleNo).toHaveBeenCalled();
});