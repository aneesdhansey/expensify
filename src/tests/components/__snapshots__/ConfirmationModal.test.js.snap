// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should display ConfirmationModal correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Testing Modal"
  isOpen={true}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title"
  >
    Testing Modal
  </h3>
  <p
    className="modal__body"
  >
    This is a test
  </p>
  <div
    className="modal__actions"
  >
    <button
      className="button button-yes"
      onClick={[MockFunction]}
    >
      Yes
    </button>
    <button
      className="button button-no button--secondary"
      onClick={[MockFunction]}
    >
      No
    </button>
  </div>
</Modal>
`;
